### PR TITLE
Add configs for email templates branding

### PR DESCRIPTION
**Pull request by amanda-ariyaratne**
### Proposed changes in this pull request

- Adds the configuration setting needed to enable/disable the capability to apply organization level branding for email templates. It is is disabled by default.
- Adds the configurations needed to define the fallbacks for branding to be used in the case where branding is enabled but no branding preferences have been provided by the tenant admin.

### Related PRs

- [#165](https://github.com/wso2-extensions/identity-event-handler-notification/pull/165)

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json [hunk 1]
```diff
@@ -891,5 +891,7 @@
   "axis2.transport_sender.https.parameters.CONNECTION_TIMEOUT": "60000",
 
   "show_pending_user_information.enable": true,
-  "organization_management.enable": false
+  "organization_management.enable": false,
+
+  "email_templates.branding.organization.enable": false
```
Comment (amanda-ariyaratne) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json:896]: remove this

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2 [hunk 2]
```diff
@@ -3179,4 +3179,18 @@
...
+
+    <!-- Configuration to set up fallback configs for organization level branding configs   -->
+    <!-- to be used in the case where branding configs are not defined by the tenant users. -->
+    <OrganizationLevelEmailBrandingFallbacks>
+        {% for key,value in email_templates.branding.fallbacks.items() %}
+            <OrganizationLevelEmailBrandingFallback key="{{key}}">{{value}}</OrganizationLevelEmailBrandingFallback>
+        {% endfor %}
+    </OrganizationLevelEmailBrandingFallbacks>
+
+
```
Comment (amanda-ariyaratne) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2:3195]: remove one empty line

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2 [hunk 3]
```diff
@@ -3179,4 +3179,18 @@
     <ShowPendingUserInformation>{{show_pending_user_information.enable}}</ShowPendingUserInformation>
     {% endif %}
 
+    <!-- Configuration to enable the organization level branding of email templates. -->
```
Comment (amanda-ariyaratne) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2:3182]: Add comment inside the if condition

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml [hunk 4]
```diff
@@ -1,9 +1,10 @@
 <?xml version="1.0" encoding="ISO-8859-1"?>
 <!--
-~ Copyright (c) 2011, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+~ Copyright (c) 2022, WSO2 LLC. (http://www.wso2.com).
```
Comment (kayathiri4) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml:3]: We should not update the year

Comment (amanda-ariyaratne) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml]: fixed with 3d40db15acfe5463579d07ae8c1dd523be4ce63f

**Review by Rashmini — APPROVED** [review]

**Pull request merged**